### PR TITLE
CSDID-1187 add remaining section 3 fields

### PR DESCRIPTION
**Pull request by ellamdav**

### CODE REVIEW

File: app/uk/gov/hmrc/cdsimportsddsfrontend/services/XmlWriter.scala [hunk 1]
```diff
@@ -27,14 +28,34 @@ trait XmlWriter[A] {
...
+      val childNodes: List[Node] = List(categoryCode, id, typeCode, lineNumeric).flattenOption
+
+      Option(childNodes).filter(_.nonEmpty).map(nonEmptyChildNodes => <PreviousDocument>{nonEmptyChildNodes}</PreviousDocument>)
+    }
+  }
+
+  implicit val authorisationHolderWriter: XmlWriter[AuthorisationHolder] = new XmlWriter[AuthorisationHolder] {
+    override def toXml(value: AuthorisationHolder): Option[Elem] = {
+      val id = maybeElement("ID", value.identifier)
+      val typeCode = maybeElement("CategoryCode", value.categoryCode)
```
Comment (retroredge) [inline app/uk/gov/hmrc/cdsimportsddsfrontend/services/XmlWriter.scala]: Is `CategoryCode` a slightly better name here?

File: app/uk/gov/hmrc/cdsimportsddsfrontend/services/XmlWriter.scala [hunk 2]
```diff
@@ -27,14 +28,34 @@ trait XmlWriter[A] {
...
+      val childNodes: List[Node] = List(id, typeCode).flattenOption
+
+      Option(childNodes).filter(_.nonEmpty).map(nonEmptyChildNodes => <AuthorisationHolder>{nonEmptyChildNodes}</AuthorisationHolder>)
+    }
+  }
+
+  implicit val domesticDutyTaxPartyWriter: XmlWriter[DomesticDutyTaxParty] = new XmlWriter[DomesticDutyTaxParty] {
+    override def toXml(value: DomesticDutyTaxParty): Option[Elem] = {
+      val id = maybeElement("ID", value.identifier)
+      val typeCode = maybeElement("RoleCode", value.roleCode)
```
Comment (retroredge) [inline app/uk/gov/hmrc/cdsimportsddsfrontend/services/XmlWriter.scala]: Is `RoleCode` a slightly better name here?

File: app/uk/gov/hmrc/cdsimportsddsfrontend/services/XmlWriter.scala (hunk 1 above)
Comment (ellamdav) [reply app/uk/gov/hmrc/cdsimportsddsfrontend/services/XmlWriter.scala]: fixed now, thanks

File: app/uk/gov/hmrc/cdsimportsddsfrontend/services/XmlWriter.scala (hunk 2 above)
Comment (ellamdav) [reply app/uk/gov/hmrc/cdsimportsddsfrontend/services/XmlWriter.scala]: fixed now, thanks

**Review by retroredge — APPROVED** [review]

**Pull request merged**